File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var minRemoveToMakeValid = function(s) {
    let arr = s.split("")
    let stack = []
    for(let index=0;index<arr.length;index++){
        if(arr[index] == "("){
            stack.push(index)
        }
        else if(arr[index]==")"){
            if(stack.length>=1){
                stack.pop()
            }
            else{
                arr[index]=""
            }
        }
    }
    if(stack.length>=1){
        stack.forEach(index=>{
            arr[index] = ""
        })
    }
    return arr.join("")
};

File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        stack = []
        sSet = set()
        for i in range(len(s)):
            if s[i]=="(":
                stack.append(i)
            elif(s[i]==")"):
                if(len(stack)>=1 and s[stack[-1]] == "("):
                    stack.pop()
                else:
                    sSet.add(i)
        ans = ""
        if(stack):
            for i in stack:
                sSet.add(i)
        for i in range(len(s)):
            if i not in sSet:
                ans+=s[i]
                
        return ans        